nservicebus/messaging/send-a-message.md

```diff
@@ -1,43 +1,54 @@
 ---
-title: Sending a Message
-summary: Describes how to send a message
+title: Sending messages
+summary: Describes how to send messages
 tags: []
 redirects:
 - nservicebus/how-do-i-send-a-message
 ---
 
-Message sending involves using Send operation that takes an argument of a message instance to be delivered. The details differ slightly between version of NServiceBus, as shown in following snippets.
+NServiceBus supports sending different types of messages (see [Messages, Events and Commands](messages-events-commands.md)) to any endpoint. Messages can be sent from outside or inside the message processing pipeline:
 
-Here's how a message is sent by directly using the instance of the endpoint
+## Outside the pipeline
+
+Messages are sent outside the pipeline when the send operation is not related to an incoming message. Examples may be:
+
+* Sending a command which is triggered by an HTML form submitted to an ASP.NET application.
+* Publishing an event after an user clicked a button on the GUI (see [Publish and Handle an Event](publish-subscribe/publish-handle-event.md)).
+
+A message is sent by directly using the instance of the endpoint:
 
 snippet:BasicSend
 
-And here's how it is done from the message handler
+Note: Up to NServiceBus version 5, `IBus` is automatically registered in the configured dependency injection container. In Version 6, `IBus` has been deprecated and replaced with `IEndpointInstance` for sends outside the pipeline. `IEndpointInstance` is not registered by default.
+
+
+## Inside the pipeline
+
+Messages are sent from within the message processing pipeline when they are sent from a message handler, a saga or some advanced extensibility points. This is code triggered by an incoming message and all send operations are linked to that incoming message. This allows transactions to rollback already sent messages when the message fails at some later stage in the pipeline.
+
+Here's how you send a simple message from a message handler:
 
 snippet:SendFromHandler
 
-In both cases you can use an interface rather than a concrete class for a message:
+It's also possible to use an interface rather than a concrete class for a message:
 
 snippet:BasicSendInterface
 
+Note: Up to NServiceBus version 5, the operations are available on the `IBus` which can be accessed using constructor or property injection. From version 6 `IMessageHandlerContext` is passed in directly to provide access to bus operations.
```

Comment:
by default we don't register `IEndpointInstance` automatically. If users to register it themselves and inject it, it means   all operations on the EndpointInstance aren't using the floated context which is present on the MessageHandlerContext. The main risk here is that the operations won't properly participate in transactions and other weird side effects like missing/wrong headers and more.